File: javascript/leetcode/509. Fibonacci Number.js
/**
 * @param {number} N
 * @return {number}
 */
var fib = function(N) {
  if (N === 0) return 0;
  if (N === 1) return 1;

  let first = 0;
  let second = 1;

  for (let i = 2; i <= N; i++) {
    let third = first + second;
    first = second;
    second = third;
  }

  return second;
};

File: go/leetcode/509. Fibonacci Number.go
func fib(N int) int {
    if N == 0 {return 0}
    if N == 1 {return 1}
    
    first := 0
    second := 1
    
    for i := 2; i <= N; i++ {
        third := first + second
        first = second 
        second = third
    }
    
    return second
}